src/System.IO/tests/BinaryWriter/BinaryWriter_WriteByteCharTests.cs

```diff
@@ -0,0 +1,571 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.IO;
+using System.Text;
+
+namespace BinaryWriterTests
+{
+    public class BinaryWriter_WriteByteCharTests
+    {
+        /// <summary>
+        /// Cases Tested:
+        /// 1) Tests that BinaryWriter properly writes chars into a stream.
+        /// 2) Tests that if someone writes surrogate characters, an argument exception is thrown
+        /// 3) Casting an int to char and writing it, works.
+        /// </summary>
+        [Fact]
+        public static void BinaryWriter_WriteCharTest()
```

Comment:
> it just occurred to me that these are ported and not your tests :)

Yes, and it's even more indirect than that.  These were tests which @FiveTimesTheFun ported, and I'm just picking up his commits from the previous PR in order to make forward progress and get these checked in.  I'll fix up the smaller issues you found, but I'll leave rewriting/augmenting some of these larger tests for a subsequent exercise.